src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -1139,4 +1140,37 @@ private int FindMetricAggregatorsCustomTag(ReadOnlySpan<KeyValuePair<string, obj
 
         return this.lookupAggregatorStore(tagKeysAndValues!, actualLength);
     }
+
+    internal sealed class ExemplarSamplingHelper
+    {
+        public bool? EarlySampleDecision;
+        public ShouldSampleFunc<long> ShouldSampleLong;
+        public ShouldSampleFunc<double> ShouldSampleDouble;
+
+        public ExemplarSamplingHelper(ExemplarFilter exemplarFilter)
+        {
+            Debug.Assert(exemplarFilter != null, "exemplarFilter was null");
+
+            if (exemplarFilter is AlwaysOffExemplarFilter)
+            {
+                this.EarlySampleDecision = false;
+                this.ShouldSampleLong = static (_, _) => false;
+                this.ShouldSampleDouble = static (_, _) => false;
+            }
+            else if (exemplarFilter is AlwaysOnExemplarFilter)
+            {
+                this.EarlySampleDecision = true;
+                this.ShouldSampleLong = static (_, _) => true;
+                this.ShouldSampleDouble = static (_, _) => true;
+            }
+            else
```

Comment:
we can special case the TraceBasedExemplarFilter too.
Infact this whole section can be removed when we do https://github.com/open-telemetry/opentelemetry-dotnet/issues/5333

We can do it separate to keep this PR's size from growing more.

Reply:
I haven't looked at filters at all yet. But I do plan to do a lot more PRs in order to get exemplars into shape 😄